samtranslator/model/api/http_api_generator.py

```diff
@@ -489,14 +489,21 @@ def _set_default_authorizer(self, open_api_editor, authorizers, default_authoriz
                 path, default_authorizer, authorizers=authorizers, api_authorizers=api_authorizers
             )
 
-    def _get_authorizers(self, authorizers_config, default_authorizer=None):
+    def _get_authorizers(self, authorizers_config, enable_iam_authorizer=False):
         """
         Returns all authorizers for an API as an ApiGatewayV2Authorizer object
         :param authorizers_config: authorizer configuration from the API Auth section
-        :param default_authorizer: name of the default authorizer
+        :param enable_iam_authorizer: if true add an "AWS_IAM" authorizer
         """
         authorizers = {}
 
+        if enable_iam_authorizer:
```

Comment:
I haven't got to the test yet but we should make sure we testing that handles different arguments. Python objects are all truthy, so what I want to make sure we validate is that the end overall experience handles things like "empty resource value, true is turned into a bool from the template, same with false, what happens when a customer sends a string instead of a bool type. 

My main concern is how this might behave. SAM runs before CFN will do anything to a template, so we get a lot of 'raw' data and can lead to small bugs that are harder to unwind/support.

Reply:
Should maybe switch to `enable_iam_authorizer == True`

Reply:
Why that over the implied `== True`? (Genuinely asking; Python is not my strong suit)

Reply:
We could pass `"False"` in this value for example and it would pass the if condition.
```
>>> a = "False"
>>> "foo" if a else "bar"
'foo'
```
We should actually use `is True` instead of `== True`:
```
>>> 1 == True
True
```

Reply:
Got it. Fairly easy change.